tests/framework/extensions/etcdsnapshot/config.go

```diff
@@ -0,0 +1,31 @@
+package etcdsnapshot
+
+import (
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+)
+
+const (
+	ConfigurationFileKey = "snapshotInput"
+)
+
+type ETCD struct {
```

Comment:
Can you explain why you can just use the native structs fro ETCD?

Reply:
I had it this way so that we can have our own `snapshotInput`, like we do for nodescaling for `scalingInput`. I didn't want to overcrowd `provisioningInput`, as we plan to have multiple etcd snapshot test cases in the future. This is just the initial PR team 2 agreed on.

Reply:
Spoke offline, modified to use native structs for ETCD and kept the provisioningInput as is.